Gems/Atom/Tools/ShaderManagementConsole/Code/Source/Window/ShaderManagementConsoleTableView.cpp

```diff
@@ -241,6 +246,79 @@ namespace ShaderManagementConsole
         // Reconnect to the notification bus now that all changes have been applied
         AtomToolsFramework::AtomToolsDocumentNotificationBus::Handler::BusConnect(m_toolId);
     }
+
+    void ShaderManagementConsoleTableView::OnOptionClicked(int column)
+    {
+        const auto& shaderOptionDescriptor = m_shaderOptionDescriptors[column];
+
+        AZStd::vector<AZ::Data::AssetId> materialAssetIdList;
+        ShaderManagementConsoleRequestBus::BroadcastResult(
+            materialAssetIdList,
+            &ShaderManagementConsoleRequestBus::Events::GetAllMaterialAssetIds);
+
+        QString materialList = "";
+
+        QProgressDialog progressDialog("Gather shader variant information...", "Cancel", 0, materialAssetIdList.size());
+        progressDialog.setWindowModality(Qt::WindowModal);
+        progressDialog.setMaximumWidth(400);
+        progressDialog.setMaximumHeight(100);
+        progressDialog.setWindowTitle("Gather information from material assets");
+
+        for (int i = 0; i < materialAssetIdList.size(); ++i)
+        {
+            AZStd::vector<AZ::RPI::ShaderCollection::Item> shaderItemList;
+            ShaderManagementConsoleRequestBus::BroadcastResult(
+                shaderItemList,
+                &ShaderManagementConsoleRequestBus::Events::GetMaterialInstanceShaderItems,
+                materialAssetIdList[i]);
+
+            for (auto& shaderItem : shaderItemList)
+            {
+                bool found = false;
+                for (auto& descriptor : shaderItem.GetShaderOptionGroup().GetShaderOptionDescriptors())
+                {
+                    if (descriptor.GetName() == shaderOptionDescriptor.GetName())
+                    {
+                        // the material is using this option
+                        AZ::IO::Path assetPath = AZ::IO::Path(AZ::RPI::AssetUtils::GetSourcePathByAssetId(materialAssetIdList[i]));
+                        materialList += QString(assetPath.Stem().Native().data());
+                        materialList += "\n";
+                        found = true;
+                        break;
+                    }
+                }
+
+                if (found)
+                {
+                    break;
+                }
+            }
+
+            progressDialog.setValue(i);
+
+            if (progressDialog.wasCanceled())
+            {
+                return;
+            }
+        }
+
+        progressDialog.close();
+
+        if (materialList != "")
+        {
+            QMessageBox msgBox;
```

Comment:
Same parenting advice as above
```suggestion
            QMessageBox msgBox(AzToolsFramework::GetActiveWindow());
```